Persistent model

igorroncevic opened the pull request:
First version of a persistent model representing our sketch of database organization.

igorroncevic committed:
Persistent model

Pajovic reviewed (commented):
Use uuid as primary key and generate it in code.
Example:
id := uuid.NewV4().String(),
uid := pgtype.UUID{}
uid.Set(id)
References:
https://pkg.go.dev/github.com/satori/go.uuid@v1.2.0
https://pkg.go.dev/github.com/jackc/pgtype@v1.4.2#UUID

Add sharing and accesing organizationIds in ext_access_rights
I suggest making Internal access rights to be bool type.

Pajovic reviewed (approved):
Approve from my side, just check comments.

igorroncevic committed:
Persistent model - changes

igorroncevic commented:
- Added sharing and receiving organization ids and updated access rights to be of type bool, according to suggestions.
- **Removed product id** from external access rights, as all products within the receiving company will be shared with same access rights initially, excluding constraints.
- **Added column for external right id** to constraints table, which will be used to identify the rule which is affected by current constraint.
